### PR TITLE
Adds back the admin verb smite also improves it

### LINKED ISSUE #112: Clowns and mimes lack maint access.
WHAT THE ACTUAL FUCK /TG/!!!!

### PR DESCRIPTION
**Pull request by KittyKathy**
:cl: Kitty Kathy
add: Ports back the admin command smite and improves it a bit
tweak: They can now smite you for clone, brain and stamina damage
/:cl:
Fixes #112
I did test this and it works.

This is a old code port as in most of it is copy pasted from old code. The lightning striking icon is the same from pre rebase except I removed the empty frames. I changed the way it shows the icon using flick instead of just changing to the icon state because that old method caused issues of the icon starting at any random frame for me when I tested it.

When this and the current butt PR gets merged I can add back farting on bibles gibing you because It uses the lightning effect this PR adds.

Adds the option to do clone damage, brain damage, stamina damage and a cancel option. Doing 0 damage would just cancel it so you can't smite someone for 0 brute/burn/oxygen/toxic/clone/brain/stamina damage.

Tell me how I can improve this and I will try to do it.

### CODE REVIEW

**Comment by tortellinitony** [comment]
Also add the effect when test dummies spawn on the old codebase

**Comment by j-mie** [comment]
Thanks :) This is 100% not a major feature but it's something I personally wanted.

**Comment by KittyKathy** [comment]
Before you merge this I am going to try and see if I can do that thing tony wanted.

File: code/modules/admin/verbs/smite.dm
Comment (ghost) [inline code/modules/admin/verbs/smite.dm]: Wouldn't this just be better as if(!dam) and removing the above 2 lines?

Comment (KittyKathy) [reply code/modules/admin/verbs/smite.dm]: I'm assuming you mean to remove lines 38 and 39 right?

Well if I did that then it would break the cancel option as only the options that do a numbered amount of damage asks for how much damage then checks if the damage entered is 0 which then cancels the smiting. So it would probably just do the animation and sound an then nothing but log something about smiting with cancel. 

I could move the check for 0 damage out of the scope where it ask for how much damage, but then it would break gib and heal because they don't use the dam variable so it would be 0 always. 

I could make line 38 a return though so it stops faster.

*edited to make it easier to read also typo.

Comment (ghost) [reply code/modules/admin/verbs/smite.dm]: I meant like this: https://ghostbin.com/paste/u5qe5

Comment (ghost) [reply code/modules/admin/verbs/smite.dm]: I see what you mean though.

**Comment by KittyKathy** [comment]
So it seems we no longer have the /turf/proc/turf_animation that the old code used to have anyone know what I should do? Add that proc back? Make a new proc just for that spawn effect only? or something else?
This is what it used before `T.turf_animation('icons/effects/96x96.dmi',"beamin",-32,0,MOB_LAYER+1,'sound/misc/adminspawn.ogg',5)`
Actually I could probably just make it similar to that lightning effect but It may not be exactly the same as it was before.

**Comment by j-mie** [comment]
I'd readd the turf_animation stuff, I ported it from /vg/ at the time.

**Comment by KittyKathy** [comment]
Okay, so I think that I'm the wrong person to add back turf_animation unless you like copy paste that might not work. I made the lightning and admin dummy spawn effect into it's own file called holy.dm that should have holy effects. I also added the holy effect from spawning yourself as a admin dummy to what the heal option does in smite, should look better now.
I tested it again with the new changes should be good to go I think. Also this might not be important but the admin dummy spawn uses the hallelujah sound that tg has which sounds a bit better than the old one to me.

I had a warning in the git terminal about LF will be replaced by CRLF in holy.dm that won't be a problem right?

**Comment by JohnGinnane** [comment]
That's caused by windows using one way of ending a line and Linux uses another. Should be okay if it sorts itself out

File: code/game/objects/effects/holy.dm
Comment (ghost) [inline code/game/objects/effects/holy.dm]: Please put new files under `modules/hippie`

File: code/game/objects/effects/holy.dm
Comment (ghost) [inline code/game/objects/effects/holy.dm]: Please use a timer for this

File: code/game/objects/effects/holy.dm
Comment (ghost) [inline code/game/objects/effects/holy.dm]: Please put this icon under `icons/hippie`

File: code/modules/admin/verbs/smite.dm
Comment (ghost) [inline code/modules/admin/verbs/smite.dm]: Clamp this to something sane

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/holy.dm
Comment (KittyKathy) [reply code/game/objects/effects/holy.dm]: Oh so all content that is ported from pre rebase or not a tg port goes there? what about smite.dm? tg did not have smite.dm so I had to add that file back.

File: code/game/objects/effects/holy.dm
Comment (KittyKathy) [reply code/game/objects/effects/holy.dm]: Are you sure about that? This icon file already existed I did not make a new file for this I just exported the icon state from the pre rebase code and imported it back into the file with the same name and a did a few edits as it had null frames.
So do you want me to leave it or make a new file something like holy.dmi with only the lightning and holy light effect icons and put it in the icons/hippie?

File: code/game/objects/effects/holy.dm
Comment (KittyKathy) [reply code/game/objects/effects/holy.dm]: So use addtimer?

File: code/modules/admin/verbs/smite.dm
Comment (KittyKathy) [reply code/modules/admin/verbs/smite.dm]: Why? this is adminbus there is no sanity here. Would this create a issue if I did not clamp it? Keep in mind this can be used on any mob I think not just humans I think you could use it on ghost too but it does not really do anything to them.
Also if they wanted they could just use variable edit to do like 1000000 burn damage, this way is just easier and more cool looking. There is probably already something that prevents humans from going over a certain damage limit like the procs it uses to do the damage.

Comment (j-mie) [reply code/modules/admin/verbs/smite.dm]: There are some mobs which can take like ~1000 damage too.

File: code/game/objects/effects/holy.dm
Comment (ghost) [reply code/game/objects/effects/holy.dm]: Yes, i fully agree with @JamieH wanting to put all sprites under their own folder, just create a new `dmi` for this specific effect or a generic `effects.dmi` in the hippie sprite folder so whenever we rebase in the future it'll be faster and more efficient

File: code/game/objects/effects/holy.dm
Comment (ghost) [reply code/game/objects/effects/holy.dm]: Yes

File: code/game/objects/effects/holy.dm
Comment (ghost) [reply code/game/objects/effects/holy.dm]: All new files should go into the hippie module IMO

**Comment by KittyKathy** [comment]
I think I did everything BTFO suggested except the clamp to sane levels thing going to try and minimize the code some more though and then test it again because this may break it not sure.

**Comment by KittyKathy** [comment]
Alright tested it and it works needs a code revive again also do you guys want me to try and fix farts in this PR or should I do it in another? I think it would be easier to do it in another PR.
There is a issue with farts where you can spam the fart macro on a bible to make it try to gib you several times causing runtimes like this.
`runtime error: `
`[14:53:45]Cannot execute null.gib().`
`[14:53:45]proc name: run emote (/datum/emote/living/carbon/human/fart/run_emote)`
`[14:53:45]  source file: emote.dm,68`
`[14:53:45]  usr: null`
`[14:53:45]  src: /datum/emote/living/carbon/hum... (/datum/emote/living/carbon/human/fart)`
`[14:53:45]  call stack:`
`[14:53:45]/datum/emote/living/carbon/hum... (/datum/emote/living/carbon/human/fart): run emote(null, null, null)`
The farting on a bible code also needs the icons this readds all it does is play the sound right now. and the superfart one trys to use the icons that don't exist in those files.

**Comment by j-mie** [comment]
Yo, it looks like /tg/ added this: https://github.com/HippieStation/HippieStation13/pull/183/commits/7e517acd6f3c93b5e73337362b7612ece59df075 - I've added it in #183 - probably needs tweaking? I dunno.

**Comment by KittyKathy** [comment]
I have to download it and see what it does, maybe close this for now and see if that system is better than the old system we had as the people that will be using this will be admins they should decide if they want the tg smite or the old smite maybe both? idk give it some time and ask them to vote on it or something.

File: code/modules/hippie/code/game/objects/effects/holy.dm
Comment (ghost) [inline code/modules/hippie/code/game/objects/effects/holy.dm]: Please use `TRUE` or `FALSE`

File: code/modules/hippie/code/modules/admin/verbs/smite.dm
Comment (ghost) [inline code/modules/hippie/code/modules/admin/verbs/smite.dm]: Please use the `addtimer` and `callback` system to do this instead of using spawns, you can find more information about this [here](https://github.com/HippieStation/HippieStation13/blob/master/code/datums/callback.dm)

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/modules/hippie/code/game/objects/effects/holy.dm
Comment (KittyKathy) [reply code/modules/hippie/code/game/objects/effects/holy.dm]: I would, but don't I have to follow whatever the parent is using? this was inherited from /obj/effect which is using 1 and 0 if I just changed this one child to use TRUE/FALSE won't it break something?

Comment (ghost) [reply code/modules/hippie/code/game/objects/effects/holy.dm]: No. `TRUE` and `FALSE` are defines for 1 and 0

File: code/modules/hippie/code/modules/admin/verbs/smite.dm
Comment (KittyKathy) [reply code/modules/hippie/code/modules/admin/verbs/smite.dm]: So you want me to make a proc for this? because the code in the spawn is not a proc and the spawn is being used as a 1 second delay so the lighting effect can hit the person then it does something which looks better.

**Comment by KittyKathy** [comment]
Anyone know how to fix this merge conflict? something with tgstation .dme
*Edit nvm I think I figured it out just need to recompile it to see if that works.

**Comment by j-mie** [comment]
build this

**Comment by KittyKathy** [comment]
If by build it you mean compile and test it, yeah I did and it works.

**Comment by JohnGinnane** [comment]
@KittyKathy this ready to be checked?

**Comment by KittyKathy** [comment]
Yeah, this is ready to be checked.

**Review by ghost — APPROVED** [review]

**Pull request merged**
